sdk/armcore/connection.go

```diff
@@ -55,25 +50,39 @@ type Connection struct {
 }
 
 // NewDefaultConnection creates an instance of the Connection type using the AzurePublicCloud.
+// Pass nil to accept the default options; this is the same as passing a zero-value options.
 func NewDefaultConnection(cred azcore.TokenCredential, options *ConnectionOptions) *Connection {
 	return NewConnection(AzurePublicCloud, cred, options)
 }
 
 // NewConnection creates an instance of the Connection type with the specified endpoint.
 // Use this when connecting to clouds other than the Azure public cloud (stack/sovereign clouds).
+// Pass nil to accept the default options; this is the same as passing a zero-value options.
 func NewConnection(endpoint string, cred azcore.TokenCredential, options *ConnectionOptions) *Connection {
 	if options == nil {
-		o := DefaultConnectionOptions()
-		options = &o
+		options = &ConnectionOptions{}
+	} else {
+		// create a copy so we don't modify the original
+		cp := *options
+		options = &cp
 	}
 	if options.Telemetry.Value == "" {
 		options.Telemetry.Value = UserAgent
 	} else {
 		options.Telemetry.Value += " " + UserAgent
 	}
+	regRPOpts := RegistrationOptions{
+		HTTPClient: options.HTTPClient,
+		Logging:    options.Logging,
+		Retry:      options.Retry,
+		Telemetry:  options.Telemetry,
+	}
+	if options.DisableRPRegistration == true {
+		regRPOpts.MaxAttempts = -1
+	}
 	p := azcore.NewPipeline(options.HTTPClient,
 		azcore.NewTelemetryPolicy(&options.Telemetry),
-		NewRPRegistrationPolicy(endpoint, cred, &options.RegisterRPOptions),
+		NewRPRegistrationPolicy(endpoint, cred, &regRPOpts),
```

Comment:
If they choose to disable RP registration, shouldnt we just remove the policy from the pipeline and avoid any additional processing? (I know it's probably very minimal, but still why add a policy that wont be used.)